### PR TITLE
feat: sort signals by criticality

### PR DESCRIPTION
**Pull request by antonbauhofer**
### Summary of changes

- add toggle state for critical signals progress bar to store
- extend sorting algorithm to consider criticality
- extend test of sorting

### Context and reason for change

There is highlighting for critical signals but before this change it was still hard to find critical signals in long lists of signals coming from e.g. ScanCode.

### How can the changes be tested

Open an input file that contains critical signals and look at the sorting.

### CODE REVIEW

**Review by MarkusObendrauf — DISMISSED** [review]
You've introduced a new `progressBarCriticalityState` but we don't actually use it anywhere besides the `TopProgressBar` itself - so what is the advantage of this over `useState<boolean>(false);`?

iirc the requirement was that we sort by criticality only if the criticality bar is selected, in which case the global state is needed.

**Comment by antonbauhofer** [comment]
> You've introduced a new `progressBarCriticalityState` but we don't actually use it anywhere besides the `TopProgressBar` itself - so what is the advantage of this over `useState<boolean>(false);`?
> 
> iirc the requirement was that we sort by criticality only if the criticality bar is selected, in which case the global state is needed.

I have fixed it, so that the sorting is only applied when the critical progress bar is toggled.

**Review by mstykow — CHANGES_REQUESTED** [review]
There are multiple high-level issues with this PR:
1. Please read the task description again. It says there that both the LOCAL and GLOBAL tabs should be sorted by criticality. At the moment, only the LOCAL tab is sorted.
2. Reading the task description, also the attribution view should sort by criticality.
3. I find it unintuitive that the progress bar toggle affects the sorting. The two seem unrelated and I doubt that I would understand this behavior, especially as a less experienced user. In addition, there is no progress bar in attribution view so that clearly the sorting can/should not depend on the toggle. Perhaps we can have a call how to handle this.

File: src/Frontend/Components/AttributionList/AttributionList.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ interface AttributionListProps {
   title: string | JSX.Element;
   topRightElement?: JSX.Element;
   filterElement?: JSX.Element;
+  sortingElement?: JSX.Element | null;
```
Comment (mstykow) [inline src/Frontend/Components/AttributionList/AttributionList.tsx:32]: i just refactored the `AttributionList` component to simplify it in another PR.
could you please rebase this PR to #2471 and then also add a test like this one here? https://github.com/opossum-tool/OpossumUI/pull/2471/files#diff-98e02d5e99b935f7b1ee96ffec1851a973d193b0430ae82865411e07780c6877R20

File: src/Frontend/web-workers/signals-worker.ts [hunk 2]
```diff
@@ -142,12 +148,19 @@ export class SignalsWorker {
...
+        'manualData',
+        'sortingByCriticalityIsActive',
+      ])
+    ) {
       this.dispatch({
         name: 'attributionsInFolderContent',
         data: getAttributionsInFolderContent({
           resourceId: this.state.resourceId,
           manualData: this.state.manualData,
+          sortByCriticality: this.state.sortingByCriticalityIsActive,
```
Comment (mstykow) [inline src/Frontend/web-workers/signals-worker.ts:163]: maybe we can also rename the attribute inside the state to have the simpler name `sortByCriticality`

**Review by mstykow — CHANGES_REQUESTED** [review]

File: src/Frontend/Components/AttributionList/AttributionList.tsx (hunk 1 above)
Comment (antonbauhofer) [reply src/Frontend/Components/AttributionList/AttributionList.tsx]: I rebased it an added an integration test as suggested.

File: src/Frontend/Components/ResourceDetailsTabs/ResourceDetailsTabs.tsx [hunk 3]
```diff
@@ -102,8 +126,19 @@ export function ResourceDetailsTabs(
...
+    'active-sorting-audit-view',
+    AuditViewSortingType.ByOccurrence,
+  );
+  const [showSortingSelect, setShowSortingSelect] = useState<boolean>(false);
+  const defaultSortingIsActive = activeSorting === defaultSorting;
+
   const dispatch = useAppDispatch();
 
+  if (!defaultSortingIsActive && !showSortingSelect) {
+    setShowSortingSelect(!showSortingSelect);
```
[the remark this answers is not in the document]
Comment (antonbauhofer) [reply src/Frontend/Components/ResourceDetailsTabs/ResourceDetailsTabs.tsx:139]: Great suggestion! Actually that simplifies it to just setting the initial value to `!defaultSortingIsActive`.

File: src/Frontend/Components/AllAttributionsPanel/AllAttributionsPanel.tsx [hunk 4]
```diff
@@ -35,6 +37,10 @@ export function AllAttributionsPanel(
   props: AllAttributionsPanelProps,
 ): ReactElement {
   const dispatch = useAppDispatch();
+  const [activeSorting] = useVariable(
```
[the remark this answers is not in the document]
Comment (antonbauhofer) [reply src/Frontend/Components/AllAttributionsPanel/AllAttributionsPanel.tsx:40]: Extracted it to utils.

File: src/Frontend/util/__tests__/get-alphabetical-comparer.test.tsx [hunk 5]
```diff
@@ -79,11 +79,39 @@ describe('getAlphabeticalComparerForAttributions', () => {
...
+  it('sorts by criticality', () => {
+    const testAttributions: Attributions = {
+      '1': {
+        packageName: 'Test package',
+        packageVersion: '1.0',
+      },
+      '2': {
+        attributionConfidence: 0,
+        comment: 'Some comment',
+        copyright: 'Copyright John Doe',
```
[the remark this answers is not in the document]
Comment (antonbauhofer) [reply src/Frontend/util/__tests__/get-alphabetical-comparer.test.tsx:97]: I replaced it with fakers, but left the basic test structure in place. I also left the additional infos in to test that those do not influence the sorting behavior.

**Pull request closed without merging**